core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/OxfordNanoporeContainer.java

```diff
@@ -0,0 +1,98 @@
+package uk.ac.bbsrc.tgac.miso.core.data.impl;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+@Entity
+@Table(name = "OxfordNanoporeContainer")
+public class OxfordNanoporeContainer extends SequencerPartitionContainerImpl {
+
+  private static final long serialVersionUID = 1L;
+
+  @ManyToOne
+  @JoinColumn(name = "flowCellVersionId")
+  private FlowCellVersion flowCellVersion;
+
+  @ManyToOne
+  @JoinColumn(name = "poreVersionId")
+  private PoreVersion poreVersion;
+
+  @Column(nullable = false)
+  @Temporal(TemporalType.TIMESTAMP)
+  private Date receivedDate;
+
+  @Temporal(TemporalType.TIMESTAMP)
+  private Date returnedDate;
+
+  public FlowCellVersion getFlowCellVersion() {
+    return flowCellVersion;
+  }
+
+  public void setFlowCellVersion(FlowCellVersion flowCellVersion) {
+    this.flowCellVersion = flowCellVersion;
+  }
+
+  public PoreVersion getPoreVersion() {
+    return poreVersion;
+  }
+
+  public void setPoreVersion(PoreVersion poreVersion) {
+    this.poreVersion = poreVersion;
+  }
+
+  public Date getReceivedDate() {
+    return receivedDate;
+  }
+
+  public void setReceivedDate(Date receivedDate) {
+    this.receivedDate = receivedDate;
+  }
+
+  public Date getReturnedDate() {
+    return returnedDate;
+  }
+
+  public void setReturnedDate(Date returnedDate) {
+    this.returnedDate = returnedDate;
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = super.hashCode();
+    result = prime * result + ((flowCellVersion == null) ? 0 : flowCellVersion.hashCode());
+    result = prime * result + ((poreVersion == null) ? 0 : poreVersion.hashCode());
+    result = prime * result + ((receivedDate == null) ? 0 : receivedDate.hashCode());
+    result = prime * result + ((returnedDate == null) ? 0 : returnedDate.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)